src/apis/paysrv.ts

```diff
@@ -0,0 +1,90 @@
+import { ajax } from 'rxjs/ajax'
+import config from '~/config'
+/* eslint-disable camelcase */
+import { ErrorDto, Status, TransactionDto, TransactionResponseDto, TransactionType } from '~/state/models/api'
+
+/*
+ * use this after a successful call to findTransactionsForBadgeNumber to calculate the outstanding dues
+ *
+ * The payment service handles all currency amounts as integers in the currency's smallest denomination, for EUR this is cents.
+ */
+export const calculateOutstandingDues = (transactions: TransactionResponseDto): bigint => {
+	let duesCents: bigint = BigInt(0)
+	let paymentsCents: bigint = BigInt(0)
+
+	transactions.payload.forEach((transaction) => {
+		if (transaction.status === Status.valid) {
+			if (transaction.transaction_type === TransactionType.due) {
+				duesCents += transaction.amount.gross_cent
+			} else {
+				paymentsCents += transaction.amount.gross_cent
+			}
+		}
+	})
+
+	return duesCents - paymentsCents
```

Comment:
> I'd prefer something like:
> 
> ```ts
> import { sum } from 'ramda'
> 
> export const calculateOutstandingDues = (transactions: TransactionResponseDto) =>
>   sum(
>     transactions.payload
>       .filter(t => t.status === 'valid')
>       .map(t => t.transaction_type === 'due' ? t.amount.gross_cent : -t.amount.gross_cent)
>   )
> ```

I couldn't directly use sum from ramda because these are bigints in my dto's. I'm not sure they need to be, probably number would be fine as data type on the frontend side wherever we have a currency amount.

Reply:
> In general I'm not entirely convinced that this needs to be part of the API module.

Me neither, but the frontend will need to check whether there are outstanding dues so it can show the "pay now" button or not. Also could add this calculation to the payment-service and provide a specific endpoint just for this purpose, but until we do ...